src/model/kaniRunner.ts

```diff
@@ -0,0 +1,185 @@
+import { execFile } from 'child_process';
+import * as fs from 'fs';
+import * as path from 'path';
+
+import * as vscode from 'vscode';
+
+import { KaniResponse } from '../constants';
+import { CommandArgs, getRootDir, splitCommand } from '../utils';
+import { responseParserInterface } from './kaniOutputParser';
+
+/**
+ * Get the system resolved path to the cargo-kani command
+ *
+ * @param kaniCommand - Full sanitized command created by kaniCommandCreate module
+ * @returns the path for the binary cargo-kani (either the installed binary or the development one)
+ */
+export function getKaniPath(kaniCommand: string): Promise<string> {
+	return new Promise((resolve, reject) => {
+		execFile('which', [kaniCommand], (error, stdout, stderr) => {
+			if (error) {
+				console.error(`execFile error: ${error}`);
+				return;
+			}
+			if (stderr) {
+				console.error(`stderr: ${stderr}`);
+				return;
+			}
+			const cargoPath = stdout.trim();
+			console.log(`Cargo is located at: ${cargoPath}`);
+
+			// Check if cargo path is valid
+			try {
+				const stats = fs.statSync(cargoPath);
+				if (stats.isFile() && path.basename(cargoPath) === kaniCommand) {
+					resolve(path.resolve(cargoPath));
+				} else {
+					reject(new Error(`Invalid kani path: ${cargoPath}`));
+				}
+			} catch (err) {
+				reject(err);
+			}
+		});
+	});
+}
+
+/**
+ * Function that runs cargo kani [args]
+ *
+ * @param kaniCommand - Full sanitized command created by kaniCommandCreate module
+ * @returns the path for the binary cargo-kani (either the installed binary or the development one)
+ */
+export async function runKaniCommand(
+	harnessCommand: string,
+	cargoKaniMode: boolean = false,
+): Promise<any> {
+	// get the full resolved path for the root directory of the crate
+	const directory = path.resolve(getRootDir());
+	const commmandSplit: CommandArgs = splitCommand(harnessCommand);
+
+	// Get cargo command and args for the command to be executed
+	const command = commmandSplit.commandPath;
+	const args = commmandSplit.args;
+	let kaniBinaryPath = '';
+
+	if (command == 'cargo' || command == 'cargo kani') {
+		const kaniBinaryPath = await getKaniPath('cargo-kani');
+		const options = {
+			shell: false,
+			cwd: directory,
+		};
+
+		return executeKaniProcess(kaniBinaryPath, args, options, cargoKaniMode);
+	} else if (command == 'kani') {
+		kaniBinaryPath = await getKaniPath(command);
+		const options = {
+			shell: false,
+		};
+
+		return executeKaniProcess(kaniBinaryPath, args, options, cargoKaniMode);
+	} else {
+		return false;
+	}
+}
+
+/**
+ * Function that returns the diff message to be displayed
+ *
+ * @param command - Full sanitized command created by kaniCommandCreate module
+ * @returns the path for the binary cargo-kani (either the installed binary or the development one)
+ */ export async function createFailedDiffMessage(command: string): Promise<KaniResponse> {
+	// Root dir of the crate and the command and args to be executed
```

Comment:
```suggestion
 */
 export async function createFailedDiffMessage(command: string): Promise<KaniResponse> {
	// Root dir of the crate and the command and args to be executed
```
Did `prettier` not catch this?